pkg/resolution/resolver/http/resolver.go

```diff
@@ -0,0 +1,218 @@
+/*
+Copyright 2023 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package http
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	resolverconfig "github.com/tektoncd/pipeline/pkg/apis/config/resolver"
+	pipelinev1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1"
+	"github.com/tektoncd/pipeline/pkg/resolution/common"
+	"github.com/tektoncd/pipeline/pkg/resolution/resolver/framework"
+)
+
+const (
+	// LabelValueHttpResolverType is the value to use for the
+	// resolution.tekton.dev/type label on resource requests
+	LabelValueHttpResolverType string = "http"
+
+	disabledError = "cannot handle resolution request, enable-http-resolver feature flag not true"
+
+	// httpResolverName The name of the resolver
+	httpResolverName = "Http"
+
+	// ConfigMapName is the http resolver's config map
+	configMapName = "http-resolver-config"
+
+	// default Timeout value when fetching http resources in seconds
+	defaultHttpTimeoutValue = "1m"
+)
+
+// Resolver implements a framework.Resolver that can fetch files from OCI bundles.
+type Resolver struct{}
+
+func (r *Resolver) Initialize(context.Context) error {
+	return nil
+}
+
+// GetName returns a string name to refer to this resolver by.
+func (r *Resolver) GetName(context.Context) string {
+	return httpResolverName
+}
+
+// GetConfigName returns the name of the bundle resolver's configmap.
+func (r *Resolver) GetConfigName(context.Context) string {
+	return configMapName
+}
+
+// GetSelector returns a map of labels to match requests to this resolver.
+func (r *Resolver) GetSelector(context.Context) map[string]string {
+	return map[string]string{
+		common.LabelKeyResolverType: LabelValueHttpResolverType,
+	}
+}
+
+// ValidateParams ensures parameters from a request are as expected.
+func (r *Resolver) ValidateParams(ctx context.Context, oParams []pipelinev1.Param) error {
```

Comment:
nit: why this is `oParams`?

Reply:
ah yeah there was two set of params at first in that function the map[string]string  and the object ones so called differently, i refactored those things before submitting it and kept the old name...